src/main/java/woowacourse/shoppingcart/service/OrderService.java

```diff
@@ -1,86 +1,69 @@
-            ordersDetailDao.addOrdersDetail(ordersId, productId, quantity);
-            cartItemDao.deleteCartItem(cartId);
-        }
-
-        return ordersId;
+        Orders orders = new Orders(orderDetails, customerId);
+        return ordersDao.save(orders);
     }
 
-    public Orders findOrderById(final String customerName, final Long orderId) {
-        validateOrderIdByCustomerName(customerName, orderId);
-        return findOrderResponseDtoByOrderId(orderId);
-    }
+    public OrderResponse findOrderById(final Long customerId, final Long orderId) {
+        validateOrderIdByCustomerId(customerId, orderId);
+        Orders orders = ordersDao.findById(orderId);
 
-    private void validateOrderIdByCustomerName(final String email, final Long orderId) {
-        final int customerId = customerDao.findByEmail(email).getId();
+        return convertOrderToResponse(orders);
+    }
 
-        if (!orderDao.isValidOrderId((long) customerId, orderId)) {
+    private void validateOrderIdByCustomerId(final Long customerId, final Long orderId) {
+        if (!ordersDao.isValidOrderId(customerId, orderId)) {
             throw new InvalidOrderException("유저에게는 해당 order_id가 없습니다.");
         }
     }
 
-    public List<Orders> findOrdersByCustomerName(final String email) {
-        final int customerId = customerDao.findByEmail(email).getId();
-        final List<Long> orderIds = orderDao.findOrderIdsByCustomerId((long) customerId);
+    public List<OrderResponse> findOrdersByCustomerId(final Long customerId) {
+        List<Orders> orders = ordersDao.findAllByCustomerId(customerId);
 
-        return orderIds.stream()
-                .map(orderId -> findOrderResponseDtoByOrderId(orderId))
-                .collect(Collectors.toList());
+        return orders.stream().map(this::convertOrderToResponse).collect(Collectors.toList());
     }
 
-    private Orders findOrderResponseDtoByOrderId(final Long orderId) {
-        final List<OrderDetail> ordersDetails = new ArrayList<>();
-        for (final OrderDetail productQuantity : ordersDetailDao.findOrdersDetailsByOrderId(orderId)) {
-            final Product product = productDao.findProductById(productQuantity.getProductId());
-            final int quantity = productQuantity.getQuantity();
-            ordersDetails.add(new OrderDetail(product, quantity));
-        }
+    private OrderResponse convertOrderToResponse(Orders orders) {
+        List<OrderProductResponse> orderProductResponses = getOrderProductResponseByOrder(orders);
+        return new OrderResponse(orderProductResponses);
+    }
 
-        return new Orders(orderId, ordersDetails);
+    private List<OrderProductResponse> getOrderProductResponseByOrder(Orders orders) {
+        return orders.getOrderDetails().stream()
+                .map(orderDetail -> new OrderProductResponse(new ProductResponse(orderDetail.getProduct()),
+                        orderDetail.getQuantity())).collect(
+                        Collectors.toList());
```

Comment:
```suggestion
                .map(orderDetail -> new OrderProductResponse(new ProductResponse(orderDetail.getProduct()),
                        orderDetail.getQuantity()))
                .collect(Collectors.toList());
```

collect 부분을 위와 같이 개행을 하면 코드 가독성이 더 좋아질 것 같아요!! 그리고 OrderProductResponse 객체를 생성할 때 외부에서 ProductResponse 등을 다 생성해서 전달하지 말고 OrderDetail 객체만 넘겨서 내부에서 생성 로직을 가지도록 하면 코드가 좀 더 간결하지 않을까요?

Reply:
`CustomerResponse`와 `OrderProductResponse`의 생성자에 도메인 객체를 전달하여 생성될 수 있도록 개선하였습니다 :)